Dockerfile

```diff
@@ -250,6 +260,8 @@ RUN curl --connect-timeout 15 --retry 5 "https://storage.googleapis.com/flutter_
   && tar xf "/tmp/flutter.xz" -C /opt/dart \
   && rm "/tmp/flutter.xz" \
   && chmod -R o+rx "/opt/dart/flutter" \
+  && rm -Rf /opt/dart/flutter/bin/cache/dart-sdk \
```

Comment:
I'm curious about these changes since they seem unrelated to arm support? I think this ends up getting deleted anyway below in the `find ... -delete` command?

Reply:
Not an expert, but it appears flutter ships with a dart-sdk, these 2 lines replace the x86 dart sdk with the arm sdk downloaded in the previous step.  I say this because without these 2 lines I get the following error during docker build:
`#26 47.08 qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory`

Reply:
Everyone is correct 🎉 :

The flutter SDK (even after https://github.com/dependabot/dependabot-core/pull/4904/files), embeds an `x86_64` dart sdk:

```
$ file /opt/flutter/new/flutter/bin/cache/dart-sdk/bin/dart 
/opt/flutter/new/flutter/bin/cache/dart-sdk/bin/dart: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, stripped
```

And the files are deleted by the find, so they aren't part of the final image.

IIUC the issue is L266 where `flutter --version` is called before the delete happens - that uses the bundled `dart-sdk` and fails, so I ninja-patched the architecture-appropriate `dart-sdk` in.

If `flutter --version` is just meant to "hello world"/verify the installation, removing that step is an alternative to this hack.

Reply:
Ah, got it, thanks! I don't think we need to make it conditional, I was just curious why it was necessary. This is worth adding a comment about?

cc: @jonassf @sigurdm to see if you have any feedback here.

Reply:
> If flutter --version is just meant to "hello world"/verify the installation, removing that step is an alternative to this hack.

I think so! When you clone flutter from github the first invocation will create the `version` file (that pub relies on) - that is why we did this step, but the packaged flutter contains it already - so I think it can just be removed!